lib/ansible/modules/unarchive.py

```diff
@@ -930,7 +930,7 @@ def unarchive(self):
     def can_handle_archive(self):
         # Prefer gtar (GNU tar) as it supports the compression options -z, -j and -J
         try:
-            self.cmd_path = get_bin_path('gtar')
+            self.cmd_path = get_bin_path('gtar', ['/usr/local/bin'])
```

Comment:
Add it conditionally on the OS being Mac OS/OS X/Darwin

Reply:
It seems like some versions probably do contain /usr/local/bin in the PATH by default... there's a path_helper script that determines the PATH from /etc/paths, which led me to https://apple.stackexchange.com/questions/215697/what-is-the-default-content-in-private-etc-paths.

The issue linked did find the ansible executable is in /usr/local/bin. (Though maybe they passed the whole path?)

Reply:
for that case the module should use basename on user input, but that is not the case here since we specifically only pass the string `gtar` and there is no user input.

Reply:
I don't think it should be corrected in target systems.

`Ansible` run commands in a non-login shell while the `/etc/paths` is loaded in a login shell.

`/etc/paths` would be loaded when run `/usr/libexec/path_helper -s` while `/usr/libexec/path_helper -s` is loaded in `/etc/profile` but it requires a login shell.

Here is what I tested

1. Add `echo "$(date): /etc/profile sourced" >> /tmp/profile_log.txt` in `/etc/profile`
2. Run `ansible example.com -m shell -a 'echo $PATH'`
3. Check the `/tmp/profile_log.txt` and see nothing
4. Run `ssh example.com`
5. Check the `/tmp/profile_log.txt` and see the log

Reference:
https://apple.stackexchange.com/questions/128897/how-to-use-etc-paths-d-to-add-executable-files-to-my-path

Reply:
that is the difference between interactive and non interactive sessions, this still is a 'configuration issue on the target', but I won't stop you from adding the path conditionally to the module ... just so people don't have to figure out one of the 10.213.098.120.938 ways to configure environment to fix this.